designs/ble-mote.stanza

```diff
@@ -118,77 +112,143 @@ pcb-module sensor-mote :
-; Set the active design directory for CAD
-set-design-directory("hardware/ble-mote")
-; Configure design to have a default 4-layer stackup and design rules.
-set-main-module(final-design)
-set-board(default-board(4, board-shape))
-set-rules(jlcpcb-rules)
-run-checks("checks.txt")
+  inst proc : processor
+  net gnd (proc.gnd)
 
-set-paper(ANSI-A4)
+  ; Add an antenna for the PICO-D4
+  if processor == ocdb/espressif/ESP32-PICO-D4/module :
+    inst ant : inverted-f-antenna
+    net (proc.ant, ant.rf-in)
+    net (gnd ant.gnd)
+    place(ant) at loc(x - 6.0, 14.0 - y, -90.0) on Top
 
-; Show the Schematic and PCB for the design
-view-board()
-view-schematic()
+  require i2c:i2c from proc
+  require pc-uart:uart() from proc
+  require pc-ctrl:gpio[2] from proc
+  net (sensors.debug, proc.uart)
+  net (sensors.power-3v3, proc.power)
+  net (sensors.humidity-i2c i2c)
+  net (sensors.particle-counter-interface.uart  pc-uart)
+  net (sensors.particle-counter-interface.set   pc-ctrl[0].gpio)
+  net (sensors.particle-counter-interface.reset pc-ctrl[1].gpio)
+
+  ; Add a BRIGHT 60.0mcd RGB led indicator to the processor (to be visible outside)
+  val rgb = add-rgb-indicator(60.0, proc, sensors.power-3v3.vdd)
+  ; Add a ground plane
+  geom(gnd):
+    copper-pour(LayerIndex(1), isolate = 0.1, rank = 1) = board-outline
+    copper-pour(LayerIndex(2), isolate = 0.1, rank = 1) = board-outline
 
-; Export CAD
-; set-export-board?(false)
-; export-cad()
+  inst logo : ocdb/jitx-logo/small(Silkscreen("artwork", Top),true)
+  place(logo) at loc(x + 1.0, 0.0 - y, 0.0) on Bottom
+
+  check-design(self)
 
-; Solve and export Bill of Materials
-; export-bom()
+; ==========================================
+; Configure the design, then run or check it
+; ==========================================
+defn run-design (run-checks?:True|False) :
+  val outline = Rectangle(30.0, 20.0)
+  val circuit = sensor-system(ocdb/espressif/ESP32-PICO-D4/module, outline)
+  set-design-directory("ble-mote")
+  set-board(ocdb/defaults/default-board(ocdb/stackups/jlcpcb-jlc2313, outline))
+  set-rules(ocdb/rules/jlcpcb-rules)
+
+  var main-module = circuit
+  if run-checks? :
+    main-module = ocdb/generator-utils/run-final-passes(circuit) ; Analyze design with a pass
+  
+  set-main-module(main-module) ; Treat the provided module as a design, and compile it.
+  
+  if run-checks? :
+    run-checks("checks.txt")
+  else:  
+    view-board()
+    view-schematic()
+
+; ====================================
+; Compile the design, view the results
+; ====================================
+defn compile-design () :
+  run-design(false)
+
+; =====================
+; Run the design checks
+; =====================
+defn check-design () :
+  run-design(true)
+
+; ===============================================
+; Export two versions of the sensor-system to CAD
+; ===============================================
+defn export-designs () :
+  set-export-backend(`altium)
```

Comment:
Could we pass an argument into this function to select which board version to export? That way the common code can be minimized. The overhead is that you then have to call the `export-designs()` function twice from somewhere else? I assume it's the REPL?

Reply:
Done.